src/Response/NeedsRedirectException.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Drupal\commerce\Response;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Form\EnforcedResponseException;
+use Drupal\Core\Routing\TrustedRedirectResponse;
+
+/**
+ * Provides an exception that represents the need for an HTTP redirect.
+ */
+class NeedsRedirectException extends EnforcedResponseException {
+
+  /**
+   * The URL to redirect to.
+   *
+   * @var string
+   */
+  protected $url;
+
+  /**
+   * The status code to use.
+   *
+   * @var int
+   */
+  protected $statusCode;
+
+  /**
+   * The headers to set.
+   *
+   * @var string[]
+   */
+  protected $headers;
+
+  /**
+   * Creates a new NeedsRedirectException object.
+   *
+   * @param string $url
+   *    The URL to redirect to.
+   * @param int $status_code
+   *    The redirect status code.
+   * @param string[] $headers
+   *    Headers to pass with the redirect.
+   */
+  public function __construct($url, $status_code = 302, array $headers = []) {
+    if (!UrlHelper::isValid($url)) {
+      throw new \InvalidArgumentException('Invalid URL provided.');
+    }
+    $this->url = $url;
+    $this->statusCode = $status_code;
+    $this->headers = $headers;
+
+    parent::__construct($this->getRedirectResponse());
+  }
+
+  /**
+   * Get the redirect URL.
+   *
+   * @return string
+   *   The URL.
+   */
+  public function getUrl() {
+    return $this->url;
+  }
+
+  /**
+   * Get the redirect status code.
+   *
+   * @return int
+   *   The status code.
+   */
+  public function getStatusCode() {
+    return $this->statusCode;
+  }
+
+  /**
+   * Get the redirect headers.
+   *
+   * @return \string[]
+   *   The array of header strings.
+   */
+  public function getHeaders() {
+    return $this->headers;
+  }
+
+  /**
+   * Get HTTP redirect response object.
+   *
+   * @return \Drupal\Core\Routing\TrustedRedirectResponse
+   *   The response object.
+   */
+  public function getRedirectResponse() {
+    return new TrustedRedirectResponse($this->getUrl(), $this->getStatusCode(), $this->getHeaders());
```

Comment:
Instead of a new class that they have to instantiate, I think I would have gone with something like this. Not *that* much better...
```
trait RedirectHelperTrait {
  protected function forceRedirect($url) {
    return new EnforcedResponseException(new TrustedRedirectResponse($url));
  }
}
```
¯\\_(ツ)_/¯